### DISCUSSION TITLE
Opening SceneViewer does not stop audio playback on previous website

### DISCUSSION THREAD

**Discussion by hybridherbst** [google/model-viewer#2190] [Q&A]
### Description

When opening SceneViewer after clicking the AR button in model-viewer, any audio that was playing on the website before opening the Intent keeps playing, and any audio that is sent along with the Intent also plays, resulting in double playback.
I'm not sure whether that's an issue with Chrome or SceneViewer (definitely not model-viewer). 
I can't imagine it being "by design" as it's very unexpected behaviour for an intent to keep playing content in the previous view.

#### Live Demo
https://glitch.com/edit/#!/rowan-righteous-shrine
Open page on Chrome Android
Click audio play button, hear train sound
Click AR button - now you have the train sound double, with different timings
Exit AR - website audio is still playing

Expected: website audio playback pauses or stops when leaving the website via Intent
Actual: website audio keeps playing

Workaround is to programmatically stop any website audio playback when user presses the AR button.

### Browser Affected

- [X] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Helios
- [ ] IE
- [ ] Safari

### OS

- [X] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by cdata** [comment]
@soraryu thanks for the report. For whatever reason, that Glitch link isn't accessible to me (but I was able to view the running demo at http://rowan-righteous-shrine.glitch.me/).

I agree with you that this isn't specifically a bug w/ `<model-viewer>`, although we might be able to make things easier for you depending on whether we can work around the issue together 🤷‍♂ 

One thing I would recommend trying is listening for the [`visibilitychange`](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event) event and then pausing or playing the audio manually based on whether `document.visibilityState === 'visible'`. Let me know if that helps address the current problem!

**Comment by hybridherbst** [comment]
Thanks, I'll try that and update here.

Also, I switched the glitch link back to non-private - I wasn't aware privacy state stays when remixing but kinda makes sense. So the link should be accessible again.

**Comment by hybridherbst** [comment]
OK, so I tried this out and document.visibilitychange event works as intended when entering/exiting AR.
Is there a way to figure out "user has successfully entered AR"? I can attach to the AR button, but that does not tell me whether entering AR actually succeeded (sometimes the button is shown but actually entering AR does not work; I think all these cases should be bugs but it still happens sometimes).

(Side note: contrary to my belief, HTML5 audio on any audio element keeps playing when switching between tabs. I think this is a change in behaviour to previous Chrome versions where only HTML audio that comes from an active Tab was allowed; not sure.  With 100+ tabs open that results in a funny search for whoever is now playing audio.)

**Comment by cdata** [comment]
Unfortunately it is not possible to know if a non-Chrome app was successfully launched from an Android intent. There are hacky tricks you could try, for example: the tab could coordinate with your file server so that the server can push an event to the tab when the `.gltf` or `.glb` file is requested by Scene Viewer. A similar hack might work for Quick Look too. In either case, it would require experimentation, and the solution would have to be tailored to your infrastructure.

Thankfully, Chrome displays a persistent notification on Android when audio is playing, so that you can stop it without going to the tab.
